scripts/post.js

```diff
@@ -332,6 +332,71 @@ function decorateImages() {
   })
 }
 
+/**
+ * Loops through a list of keywords and looks for matches in the article text.
+ * The first occurrence of each keyword will be replaced with a link.
+ */
+async function addInterLinks() {
+  const response = await fetch('/en/keywords.json');
+  if (response.ok) { 
+    const json = await response.json();
+    let bStart = Date.now();
+    const keywords = (Array.isArray(json) ? json : json.data)
+      // scan article to filter keywords down to relevant ones
+      .filter(({ Keyword }) => document.querySelector('main').textContent.toLowerCase().indexOf(Keyword) !== -1);
+    console.log(`keywords: filtering down from ${json.length} to ${keywords.length} took ${Date.now() - bStart}ms`);
+
+    // find exact text node matches and insert links (exclude headings and anchors)
+    bStart = Date.now();
+    document.querySelectorAll('main > div :not(h1):not(h2):not(h3):not(h4):not(h5):not(a)').forEach((p, num) => {
+      if (keywords.length === 0) return;
+      const textNodes = Array.from(p.childNodes)
+        // filter out non text nodes  
+        .filter(node => node.nodeType === Node.TEXT_NODE)
+        .forEach((textNode) => {
+          const matches = [];
+          const deduper = [];
+          // find case-insensitive matches inside text node
+          keywords.forEach((item, index) => {
+            const match = new RegExp(`\\b(${item.Keyword})\\b`, 'iu').exec(textNode.nodeValue);
+            if (match && !deduper.includes(match.index)) {
+              matches.push({
+                item,
+                start: match.index,
+                end: match.index + item.Keyword.length,
+              });
+              deduper.push(match.index);
+            }
+          });
+          matches
+            // sort matches descending
+            .sort((a, b) => {
+              return a.start < b.start ? 1 : -1;
+            })
+            // split text node and insert link with matched text
+            .forEach(({ item, start, end }) => {
+              const text = textNode.nodeValue;
+              const a = createTag('a', {
+                title: item.Keyword,
+                href: item.URL,
+              });
+              a.appendChild(document.createTextNode(text.substring(start, end)));
+              p.insertBefore(a, textNode.nextSibling);
+              p.insertBefore(document.createTextNode(text.substring(end)), a.nextSibling);
+              textNode.nodeValue = text.substring(0, start);
```

Comment:
now, you messed up the `textNode.nodeValue`, so your start/end need to be adjusted

Reply:
That's why I'm sorting the matches in descending start order, so replacements happen from right to left inside the text node.

Reply:
ok. 2 edge cases:
- text starts with a keyword. so you end up with an empty text node
- overlapping keywords. eg, text: "the quick brown fox".  keyword 1: "quick brown", keyword 2: "brown fox".

Reply:
The first case should actually work: start would be `0`, so that keyword would be the last replacement. 
But you might be on to something with the second case: I ran into some cases (e.g. [here](https://interlink--theblog--adobe.hlx.page/en/publish/2020/08/31/merkle-ceo-shifting-marketing-budgets-priorities-covid19.html) where one keyword was a substring of another. I thought this particular problem would best be solved by educating authors to take care of the keyword order in the spreadsheet: always put more specific keywords before less specific ones to make sure the earlier get handled first, i.e.:
1. Customer Experience Management
2. Customer Experience
I'll definitely do some testing with section overlapping keywords.

Reply:
you could try to detect overlapping ranges and then keep the longer one.